src/Uno.UI/UI/Xaml/Controls/Border/BorderLayerRenderer.skia.cs

```diff
@@ -197,20 +204,32 @@ private static IDisposable InnerCreateLayer(UIElement owner, LayoutState state)
 				}
 				else if (background is ImageBrush imgBackground)
 				{
-					//var bgSrc = imgBackground.ImageSource;
-					//if (bgSrc != null && bgSrc.TryOpenSync(out var uiImage) && uiImage.Size != CGSize.Empty)
-					//{
-					//	var fullArea = new Rect(
-					//			area.X + borderThickness.Left,
-					//			area.Y + borderThickness.Top,
-					//			area.Width - borderThickness.Left - borderThickness.Right,
-					//			area.Height - borderThickness.Top - borderThickness.Bottom);
+					imgBackground.Subscribe(imageData => {
+
+						if (imageData.Error is null)
+						{
+							var surfaceBrush = compositor.CreateSurfaceBrush(imageData.Value);
 
-					//	var insideArea = new Rect(CGPoint.Empty, fullArea.Size);
-					//	var insertionIndex = 0;
+							var sourceImageSize = new Size(imageData.Value.Image.Width, imageData.Value.Image.Height);
+
+							// We reduce the adjustedArea again so that the image is inside the border (like in Windows)
+							var imageArea = adjustedArea.DeflateBy(new Thickness((float)adjustedStrokeThicknessOffset));
+
+							backgroundArea = imgBackground.GetArrangedImageRect(sourceImageSize, imageArea);
+
+							// surfaceBrush.Offset = new Vector2((float)imageFrame.Left, (float)imageFrame.Top);
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5932478)